oeps/processes/oep-0055-proc-project-maintainers.rst

```diff
@@ -0,0 +1,428 @@
+responsible for the health of the component and the component's community.
+The remainder of this specification will be focused on the details of
+the role.
+
+The term **core contributor** will mean a named person participating
+in the `Core Contributor Program`_ as a coding core contributor and
+meeting the requirements of that program.
+
+The term **maintainer scrum-of-scrums** will mean the group of all
+maintainers who are responsible for flagging issues to project
+leadership and proposing process improvements for component
+maintenance.
+
+The term **Technical Oversight Committee** (TOC) will mean the group
+responsible for the overall oversight and stewardship of the Open edX
+project. The TOC's function with respect to the project is to be the
+arbiter of last resort for governance matters. The TOC itself was established
+and is governed by a formal
+charter.  You can read the full document `here
+<https://openedx.org/wp-content/uploads/2021/12/TCRIL-Technical-Oversight-Committee-Charter-rev.11-16.21.pdf>`_.
+
+Historical Context
+------------------
+Historically, the Open edX project has not had a well-defined role
+of maintainer.  At times, community members stepped into this role
+informally.  The role was also part of a broader "owner" role at edX.
+Owners were also responsible for the health of the component they
+owned both in the repository and in production; owners answered the
+pager when something went wrong.
+
+What Maintainers Do
+-------------------
+The role of the repository or project maintainer is both a community
+focused and a technical role [#role]_.  Maintainers will be the public face
+of each component and should be welcoming and available.  While
+maintainers should strive to always be friendly and kind in their
+communication, firmness and directness are equally important.
+Maintainers must enforce the project's standards.  In cases where a
+proposed change isn't aligned with the our standards or direction, it
+is best to make that clear as early as possible.
+
+Non-technical responsibilities of maintainers include:
+
+* Effective communication with community members, other maintainers,
+  and project leadership
+* Selection and approval of Core Contributors with commit rights to their repositories
+* Disciplined management to ensure that maintenance work is done
+* Enforcement of the project's standards, both technical and social: maintainers
+  must remain friendly, kind, and open, but that should not supercede
+  enforcement of technical standards
+* The ability to solve or escalate issues facing users of the
+  component
+
+Technical responsibilities of maintainers include:
+
+* Understanding the current component architecture and implementation
+* Ensuring that basic documenation for the component exists and is
+  accurate
+* Having a documented technical vision for the evolution of the
+  component
+* The ability to assess the appropriateness of proposed changes to or
+  extensions of the component
```

Comment:
NB: This is related to one of the items of feedback from a core contributors sprint retrospective - see section _"2. Getting feedback/reviews for OSPRs is as always a challenge"_ from https://discuss.openedx.org/t/core-contributor-sprint-retro-march-4-18th/6876/4

Reply:
> commit to provide prompt reviews for changes

On one hand, this would be an amazing state to end up in. On the other hand, from my experience on a team that "maintained" a couple dozen repos, reviewing a steady stream of PRs from outside my team felt like an enormous and overwhelming task.

Personally, I would be happy if maintainers promptly _triaged_ all incoming PRs (which might include giving a time estimate and/or advertising the PR to other willing reviewers) and committed to reviewing a finite number of PRs per maintained repo per period of time (strawman example: "maintainers should triage all incoming PRs, and themselves review an average of >=1 PR per component per month"). Committing maintainers to entirely emptying their review backlog, though, might be too big of an ask at this point in time.

Reply:
> commit to provide prompt reviews for changes

On one hand, this would be an amazing state to end up in. On the other hand, from my experience on a team that "maintained" a couple dozen repos, reviewing a steady stream of PRs from outside my team felt like an enormous and overwhelming task.

Personally, I would be happy if maintainers promptly _triaged_ all incoming PRs (which might include giving a time estimate and/or advertising the PR to other willing reviewers) and committed to reviewing a finite number of PRs per maintained repo per period of time (strawman example: "maintainers should triage all incoming PRs, and themselves review an average of >=1 PR per component per month"). Committing maintainers to entirely emptying their review backlog, though, might be too big of an ask at this point in time.

Reply:
@antoviaque @kdmccormick I agree with both of you. I have personally always been a fan of closing PRs if review was not going to happen to them, which to me does fall under "review promptly" (as a person on the other end of a PR, it's so frustrating to _not know_ and/or get pinged six months after you've dropped context... I'd rather just get a No). But Kyle's suggestion of "triage" is much nicer. Can we agree on this wording?

> The ability to promptly triage incoming requests that propose changes to or extensions of the component, assessing their appropriateness and/or routing them to proper reviewers

Reply:
Works for me 👍🏻

Reply:
@sarina Sounds good to me as a first step -- this way we'll get to see what happens, and having a reliable and fast triage would be a great first step. If there are still delays with _actually_ getting contributions reviewed afterwards though, I would require actual reviews from maintainers within a given timeframe (2 weeks to start?). As an open source project, being able to promptly review new contributions is important - we won't be a mature project until we can offer it reasonably reliably and promptly.